lpc55_sign/src/verify.rs

```diff
@@ -403,23 +406,60 @@ fn check_signed_image(image: &[u8], cmpa: CMPAPage, cfpa: CFPAPage) -> Result<bo
             3 => rkth_revoke.rotk3,
             _ => unreachable!("rkh_table must be exactly 4 elements"),
         };
-        if rotk_status == ROTKeyStatus::Invalid {
-            error!("RKH table has revoked this root certificate");
-            failed = true;
-        } else {
-            okay!("RKH table has enabled this root certificate");
+        match rotk_status {
+            ROTKeyStatus::Invalid => {
+                error!("RKH table slot {index} is disabled in CFPA");
+                failed = true;
+            }
+            ROTKeyStatus::Enabled => {
+                okay!("RKH table slot {index} is enabled in CFPA");
+            }
+            ROTKeyStatus::Revoked1 | ROTKeyStatus::Revoked2 => {
+                error!("RKH table slot {index} has been revoked in CFPA");
+                failed = true;
+            }
         }
     } else {
-        error!("Certificate 0's public key is not in RKH table");
+        error!("Root certificate's public key is not in RKH table");
+        failed = true;
+    }
+
+    let last_cert = &certs.last().unwrap();
+
+    let last_cert_sn = last_cert.tbs_certificate.serial.to_bytes_be();
+    let last_cert_sn_magic = &last_cert_sn[0..2];
+    if last_cert_sn_magic != [0x3c, 0xc3] {
+        error!(
+            "Last certificate's serial number has wrong magic prefix.  Expected 0x3cc3.  Found 0x{}",
+            last_cert_sn_magic.encode_hex::<String>()
+        );
         failed = true;
+    } else {
+        okay!("Verified last certificate's serial number has correct magic prefix");
+    }
+
+    let last_cert_sn_revoke_id = u16::from_le_bytes(last_cert_sn[2..4].try_into().unwrap());
+    if last_cert_sn_revoke_id != last_cert_sn_revoke_id.next_power_of_two() - 1 {
+        warn!("Last certificate's revocation ID (0x{last_cert_sn_revoke_id:04x}) should be a uniary counter but isn't")
+    }
+
+    let cfpa_image_key_revoke = (cfpa.image_key_revoke & 0xFFFF) as u16;
+    match last_cert_sn_revoke_id {
+        x if x == cfpa_image_key_revoke => okay!(
+            "Verified last certificate's revocation ID (0x{last_cert_sn_revoke_id:04x}) matches CFPA IMAGE_KEY_REVOKE"
+        ),
+        x if x == cfpa_image_key_revoke + 1 => okay!(
```

Comment:
I'm suspicious that this isn't `cfpa_image_key_revoke + 1`, but rather `(cpfa_image_key_revoke << 1) | 1`; i.e. the next revocation ID, not the next integer value.